### ISSUE TITLE
Failing to compile on ARM device

### ISSUE THREAD

**Issue by kirkins** [flier/rust-macho#11]
I'm relatively new to Rust, mainly just using it to compile a program I need on an ARM device.

The program requires `mach_object` which seems to cause the error:

```
   Compiling parking_lot v0.5.1
error[E0308]: mismatched types
   --> /home/rescue/.cargo/registry/src/github.com-1ecc6299db9ec823/mach_object-0.1.5/src/commands.rs:694:54
    |
694 |         unsafe { Ok(String::from(try!(CStr::from_ptr(buf.as_ptr() as *const i8).to_str()))) }
    |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

   Compiling string_cache_codegen v0.4.0
error[E0308]: mismatched types
   --> /home/rescue/.cargo/registry/src/github.com-1ecc6299db9ec823/mach_object-0.1.5/src/commands.rs:889:54
    |
889 |         unsafe { Ok(String::from(try!(CStr::from_ptr(s.as_ptr() as *const i8).to_str()))) }
    |                                                      ^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error: aborting due to 2 previous errors

error: Could not compile `mach_object`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by kirkins** [comment]
@mitsuhiko I saw your [commit](https://github.com/flier/rust-macho/commit/c5eaf1bf1626a47b10b2fab9d2ea630e486c1f56) for Windows support. I'm also here in regards to `sentry-cli`.

I'm wondering if the solution for my ARM device might be similar to your Windows issue. I'm completely new to Rust.

**Comment by mitsuhiko** [comment]
Sounds like `c_char` has different types on ARM. So presumably casting to `c_char` instead of `i8` is going to fix that? Unsure though. Definitely different issue.

**Comment by FenrirWolf** [comment]
C's `char` type can be either signed or unsigned (`i8` or `u8`) depending on the target platform. So changing those functions to use `c_char` should indeed fix the ARM issue.

**Comment by flier** [comment]
Interesting, PR merged and bump version to 0.1.6
